ybdata/validate.py

```diff
@@ -128,7 +128,7 @@ def standard_package_checklist(path, include_optional=False):
     # Data match checking (e.g. same shapes)
     if meta and exists_in_dataset(path, name+".csv.gz") and exists_in_dataset(path, name+".npz"):
         # Check numpy data structures
-        nf = np.load(os.path.join(path, name+".npz"), allow_pickle=False)
+        nf = np.load(os.path.join(path, name+".npz"), allow_pickle=True)
```

Comment:
Unfortunately, we cannot have `allow_pickle=True` for the Yellowbrick datasets -- it presents too great a security risk and causes our package to get flagged in various security warnings on GitHub and PyPI. You're right, the problem is the storage of strings and other Python objects. My solution for other datasets was to encode them with a different dtype in the `.npz` array; and you have a couple of options here. 

1. Make sure that your numeric data is encoded in numpy datatypes, e.g. `"i4"` (int32) or `"f8"` (float64') 
2. Encode any categorical variables with dummies and list those dummies in the meta. 
3. Use `u18` or a specific length unicode encoding for player names rather than strings 
4. or simply omit player names from the numpy arrays - they likely won't be a feature used in analysis, but they can stay in the CSV file. 

Since you have a heterogenous dataset, using the `--X-dtype` and `--y-dtype` flags (from above) probably won't help; you'll have to manually encode the `X` and `y` arrays before saving them as a `.npz` file.

Reply:
Appreciate the list of options! I ended up making dummies for the position feature and then lumped all other string features into an "extra_text_fields" section in the metadata. The validate step now works while leaving `allow_pickle=False` to remain as is.

Reply:
Glad to hear it - so this means that the `.npz` file contains no text fields but the CSV does?

Reply:
Yep, that's correct.